# http://jsideas.net/ pandas tutorial

In [2]:
import pandas as pd 

data = pd.read_csv('C:/Users/Dongwoon/Dropbox/공부할거/코딩/파이썬/pandas/convenient_store.csv')   #csv를 dataframe 형태로 불러옴
data.head(10)    # 괄호 안에 있는 수 만큼 row 보여줌
data.info()      # data 정보 알려줌

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 177 entries, 0 to 176
Data columns (total 7 columns):
area           177 non-null object
company        177 non-null object
hourly_wage    177 non-null int64
area1          177 non-null object
area2          177 non-null object
outlier        177 non-null int64
name           177 non-null object
dtypes: int64(2), object(5)
memory usage: 9.8+ KB


- class 'pandas.core.frame.DataFrame' : data의 class는 dataframe임
- 177 entries : row(observation) = 177
- Data columns : column 이름들 
- object, int64 : data type

In [3]:
data.area          #area column의 data 줘
data.company       #company column의 data 줘
data.describe()    #숫자인 친구들 간단한 통계값 보여줘
data.hourly_wage.describe()  #hourly_wage column의 간단한 통계값 보여줘

count     177.000000
mean     5787.627119
std       352.318646
min      5580.000000
25%      5580.000000
50%      5600.000000
75%      6000.000000
max      7500.000000
Name: hourly_wage, dtype: float64

In [4]:
data[data.hourly_wage > 6000]    # data에 필터 적용, dataframe이라서 [] 사용
data[((data.area1 == '마포구') | (data.area1=='강남구')) & (data.hourly_wage > 6000)]  # 괄호, &, | 로 복잡한 필터
cu = data[data.company.str.contains('CU')]

In [5]:
data['Seoul'] = 'in seoul'  #새 column 'Seoul' 추가

In [14]:
def more_than_6000(x):      #함수 하나 정의해보자
  if x > 6000:
    return 1
  else:
    return 0

data['more_than_6000'] = data.hourly_wage.map(lambda x: more_than_6000(x))

map(f,x) : f(x)를 return<br />
따라서 data['more_than_6000'] = data.hourly_wage.map(lambda x: more_than_6000(x)) 해석해보면
- more_than_6000 column을 만들껀데, 
- data.hourly_wage랑 map 할거다(x로 data.hourly_wage를 쓸거다)
- 이 때 f로는 lambda 써서 map 해라

In [15]:
data   #새로운 column 추가 확인

area                  company  hourly_wage area1 area2  outlier  \
0      강남구 삼성동               gs25 오크우드점         5600   강남구   삼성동        0   
1      강남구 삼성동                gs25 코엑스점         5700   강남구   삼성동        0   
2      강서구 화곡동                  gs25편의점         5600   강서구   화곡동        0   
3      광진구 군자동                     gs25         5580   광진구   군자동        0   
4      광진구 중곡동                gs25중곡대원점         5580   광진구   중곡동        0   
5      구로구 구로동                gs25구로동양점         6000   구로구   구로동        0   
6      구로구 구로동                gs25구로동양점         5580   구로구   구로동        0   
7     동대문구 장안동                gs25장안중앙점         5600  동대문구   장안동        0   
8      마포구 서교동               gs25 홍대아트점         5600   마포구   서교동        0   
9     성동구 금호동4            gs25 서울숲푸르지오점         6000   성동구  금호동4        0   
10    성북구 동소문동                gs25동소문본점         5580   성북구  동소문동        0   
11    성북구 하월곡동             gs25 성북 푸르지오         5800   성북구  하월곡동        0   
12     송파구 송파동                gs25송파중앙점         5600   송파구   송파동        0   
13   영등포구 영등포동                     gs25         6300  영등포구  영등포동        0   
14     은평구 신사동                     gs25         5580   은평구   신사동        0   
15     종로구 관수동                gs25 국일관점         5600   종로구   관수동        0   
16      중구 소공동                     gs25         5600    중구   소공동        0   
17      중구 신당동                     gs25         5580    중구   신당동        0   
18     강남구 논현동             세븐일레븐 논현11호점         6200   강남구   논현동        0   
19     강남구 신사동    편의점/ 세븐일레븐 압구정 리갈펠리스점         5600   강남구   신사동        0   
20     강남구 논현동              세븐일레븐 논현제일점         5700   강남구   논현동        0   
21    강남구 대치4동              세븐일레븐 대치6호점         5580   강남구  대치4동        0   
22     강남구 역삼동                    세븐일레븐         5800   강남구   역삼동        0   
23     강남구 논현동                    세븐일레븐         5580   강남구   논현동        0   
24     강남구 역삼동                    세븐일레븐         5580   강남구   역삼동        0   
25     강남구 대치동                    세븐일레븐         5580   강남구   대치동        0   
26     강남구 역삼동             세븐일레븐 강남KT지점         5580   강남구   역삼동        0   
27     강남구 대치동              세븐일레븐 대치3호점         6000   강남구   대치동        0   
28     강남구 역삼동            세븐일레븐 역삼 13호점         5600   강남구   역삼동        0   
29     강남구 논현동                 세븐일레븐편의점         7500   강남구   논현동        0   
..         ...                      ...          ...   ...   ...      ...   
147    강서구 방화동                 CU방화오피스점         6000   강서구   방화동        0   
148    관악구 신원동                  CU신림인석점         5580   관악구   신원동        0   
149    관악구 봉천동                  CU관악상상점         5580   관악구   봉천동        0   
150    구로구 구로동         CU편의점(고대구로병원1호점)         5580   구로구   구로동        0   
151    금천구 독산동                CU 독산메이퀸점         6300   금천구   독산동        0   
152    금천구 가산동          뚜레쥬르,CU,가산디폴리스점         6300   금천구   가산동        0   
153   노원구 하계1동               CU과학기술대원룸점         5580   노원구  하계1동        0   
154    노원구 상계동                 CU/상계주공점         5580   노원구   상계동        0   
155    노원구 공릉동                 CU공릉과기대점         5580   노원구   공릉동        0   
156    동작구 상도동                   CU 편의점         5800   동작구   상도동        0   
157    마포구 창전동                 CU 홍대으뜸점         5900   마포구   창전동        0   
158    마포구 동교동                 CU 동교행복점         5600   마포구   동교동        0   
159    서대문구 합동              CU 충정로 프랑스점         5580  서대문구    합동        0   
160   서대문구 신촌동  BGF Retail CU 신촌세브란스병원점         6900  서대문구   신촌동        0   
161   서대문구 현저동                CU 독립문극동점         6500  서대문구   현저동        0   
162  서대문구 충정로2             충정로 골든타워점 CU         6000  서대문구  충정로2        0   
163   서대문구 홍은동                  CU홍은두산점         5580  서대문구   홍은동        0   
164   서초구 반포4동                 CU 서래중앙점         6000   서초구  반포4동        0   
165    성동구 용답동                CU  장한평역점         5580   성동구   용답동        0   
166   성동구 성수동1                 CU 성수사랑점         6000   성동구  성수동1        0   
167   

In [44]:
data2 = data[(data.hourly_wage>6300)&(data.name=='CU')][['area1', 'hourly_wage']]  # 불러올 조건 지정 + 불러올 column 지정
data2

area1  hourly_wage
145   강남구         6500
160  서대문구         6900
161  서대문구         6500
172   용산구         6600
173   용산구         7500

In [45]:
data2.to_csv('data2.csv', index=False)    # index=False : dataframe의 index column(첫 번째 column)은 뺄거야